eosio.unregd/README.md

```diff
@@ -0,0 +1,58 @@
+![Alt Text](https://i.imgur.com/6F5aHWH.png)
+
+# Build
+
+eosiocpp -o eosio.unregd.wast eosio.unregd.cpp
+
+# Setup
+
+```./setup.sh```
+
+The setup script will install one contract (besides the defaults ones):
+  
+  `eosio.unregd` (empty)
+
+You need to have nodeos running.
+
+# Dependecies
+
+ ```pip install bitcoin --user```
+ ```pip install requests --user```
+ ```sudo apt-get install python-pysha3```
+
+# Test
+
+```./test.sh```
+
+The test step will:
+
+ 0. Generate a new ETH address with a random privkey.
+ 
+ 1. Add the ETH address (`eosio.unregd::add`) and transfers a random amount of 
+    EOS to the `eosio.unregd` contract.
+    This is to simulate a user that contributed to the ERC20 but 
+    didn't register their EOS pubkey.
+
+ 2. Call `eosio.unregd::regaccount` with:
+
+	* A signature for the message "$lib_block_num,$lib_block_prefix" generated with the ETH privkey
```

Comment:
I think this message is insufficient.  Unless I've missed some other secure commitment to it, this allows an attacker to copy the signature parameter and use the same LIB/tapos information to forge a competing claim that would route funds to a different account and/or eos-pubkey.

at a minimum this message should include the eos-pubkey.  I think it should include the account name as well but so long as the rightful owner has control that is less of an issue.